src/backend/commands/extension.c

```diff
@@ -2918,7 +2919,7 @@ ExecAlterExtensionStmt(AlterExtensionStmt *stmt)
  * scripts don't break if newer versions have different control parameters.
  */
 static void
-ApplyExtensionUpdates(AlterExtensionStmt *stmt, Oid extensionOid,
+ApplyExtensionUpdates(Oid extensionOid,
```

Comment:
And what was the reason of refactoring this in current patch?

Reply:
This function is supposed to accept `AlterExtensionStmt` variable in 5x because it's used in `ExecAlterExtensionStmt` only. But in 6x it's used in `CreateExtension` as well and doesn't accept any statement at all. `AlterExtensionStmt` and `CreateExtensionStmt` don't have any base type so there's nothing to cast to.

Reply:
> `AlterExtensionStmt` and `CreateExtensionStmt` don't have any base type so there's nothing to cast to.

They both Nodes, at least.

I don't see any single comment (in the right place) about importance of refactoring.

Reply:
Actually, why do we consider this as refactoring? `ApplyExtensionUpdates` doesn't have this arg in 6x, so this is a straight port of this code base. Also, if we want to keep this argument we need to change this line anyway. Why not just remove it to avoid any confusions? I think this diff can be really confusing considering that this argument is not used in the function at all, but we still keep it

Reply:
Description added